src/hott/06-contractible.rzk.md

```diff
@@ -142,8 +142,7 @@ A type is contractible if and only if its terminal map is an equivalence.
   ( A : U)
   (e : terminal-map-is-equiv A)
   : is-contr A
-  := ( (first (first e)) unit ,
-       (second (first e)))
+  := ( (first (first e)) unit , (second (first e)))
```

Comment:
Why not something like

```rzk
  : terminal-map-is-equiv A → is-contr A
  := \ ((f , is-retract-f) , _) → ( f unit , is-retract f )
```

The point being to use pattern matching for pairs instead of `first`/`second`.

Reply:
I agree that `first (first e)` and so on is not very readable. However, my preferred remedy for this would be to define named projection mappings `map-equiv`, `is-retraction-map-equiv` and so on, similar to how the projections `center-contraction` and `homotopy-contraction` are defined. I don't know how usable this is in `rzk` however, given that you don't have implicit arguments. This might make the code excessively verbose.

Reply:
Also note that having named projections has the advantage that you don't have to modify the type signature of your definitions, and is agnostic to which order the arguments of your construction come in.

Reply:
Regardless, I consider this outside the scope of this PR, although I definitively agree that it is a problem.

Reply:
Yes, projection mappings should work as well. This is not an urgent matter, we can wait for implicit arguments, which, hopefully, will be there sometime soon :)

Reply:
This makes me very excited!